anydex/wallet/stellar/xlm_provider.py

```diff
@@ -0,0 +1,158 @@
+        """
+        Get the sequence number of an address
+
+        :param address: address to get the sequence number of
+        :return: sequence number
+        """
+
+    @abc.abstractmethod
+    def submit_transaction(self, tx):
+        """
+        submit a transaction to the stellar netowrk.
+
+        :param tx: base64 xdr encoded transactions
+        :return: the transactin hash
+        """
+
+    @abc.abstractmethod
+    def check_account_created(self, address) -> bool:
+        """
+        check if an account is created or not.
+
+        :param address: address of account to check
+        :return: true or false if account is created or not
+        """
+
+
+class HorizonProvider(StellarProvider):
+    """
+    Horizon is a software that allows you to query nodes via http.
+    This should be the only thing we need since horizon also indexes the data it holds.
+    """
+
+    def __init__(self, horizon_url='https://horizon-testnet.stellar.org/'):
+        self.server = Server(horizon_url=horizon_url)
+
+    def _make_request(self, fun, *args, **kwargs):
+        """
+        Used to make the request to the horizon server.
+        It will catch the library specific exceptions and raise the exceptions used in this project.
+        :param fun: api call to make
+        :param args: args for the api call
+        :param kwargs: args for the api call
+        :return: response form the api call
+        """
+        try:
+            return fun(*args, **kwargs)
+        except stellar_sdk.exceptions.ConnectionError:
+            raise ConnectionException('Could not connect to the server')
+        except stellar_sdk.exceptions.NotFoundError as e:
+            # This exception is used by the check_account_created function
+            if fun == self.server.load_account:
+                raise e
+            raise RequestException('Check the request params, the resource could not be found')
+        except stellar_sdk.exceptions.BadRequestError:
+            raise RequestException('Your request had an error')
+        except stellar_sdk.exceptions.BadResponseError:
+            raise RequestException('The server response had an error')
+
+    def submit_transaction(self, tx):
+        return self._make_request(self.server.submit_transaction, tx)['hash']
+
+    def get_balance(self, address):
+        # We only care about the native token right now.
+        return self._make_request(self.server.accounts().account_id(address).call)['balances'][0]['balance']
+
+    def get_transactions(self, address):
+        response = self._make_request(
+            self.server.transactions().for_account(address).limit(200).include_failed(True).call)
+        transactions = response['_embedded']['records']
+        return self._normalize_transactions(transactions)
+
+    def _normalize_transactions(self, transactions):
+        """
+        Transform a list of Transactions from the api into the format useed in this project
```

Comment:
Just a typo to point out. Also, nice functionality to normalize a single transaction or a list of them. Time saver down the line